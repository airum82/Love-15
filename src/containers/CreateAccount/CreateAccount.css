.create-account {
  margin: 0 auto;
  text-align: center;
}

.create-account form {
  width: 55%;
  margin: 5% auto;
}

.create-account button {
  background-color: #60b0f4;
  border: none;
  color: white;
  display: block;
  margin: 2% auto;
  height: 5vh;
  width: 23vw;
}

.create-account input {
  height: 4vh;
  width: 40%;
}