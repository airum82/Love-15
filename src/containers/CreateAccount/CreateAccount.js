import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { createAccount } from '../../actions';
import { withRouter } from 'react-router-dom';
import './CreateAccount.css';
import { auth } from '../../firebase';

export class CreateAccount extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      email: '',
      password: '',
      dateOfBirth: ''
    }
  }

  handleDataEntry = (e) => {
    this.setState({
      [e.target.name] : e.target.value
    })
  }

  resetState = () => {
    this.setState({
      name: '',
      email: '',
      password: '',
      dateOfBirth: ''
    })
  }

  render() {
    return (
      <div className='create-account'>
        <form 
          onChange={this.handleDataEntry}
          onSubmit={(e) => {
            auth.doCreateUserWithEmailAndPassword(
              this.state.email,this.state.password)
              .then(authUser => this.props.handleCreateAccount(this.state))
              .catch(error => this.setState({ error: error.message }))
            this.resetState();
            e.preventDefault();
            this.props.history.push('/');
            console.log(this.state)
          }}
        >
          <p>Email: </p><input type="email" name="email" value={this.state.email} />
          <p>Name: </p><input type="text" name="name" value={this.state.name} />
          <p>Passworld: </p><input type="password" name="password" value={this.state.password} />
          <p>DOB: </p><input type="text" name="dateOfBirth" value={this.state.dateOfBirth} />
            <button>Create Account</button>
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  handleCreateAccount: (accountInfo) => dispatch(createAccount(accountInfo))
})

export default withRouter(connect(null, mapDispatchToProps)(CreateAccount));