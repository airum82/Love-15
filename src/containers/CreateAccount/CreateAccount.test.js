import { shallow, mount} from 'enzyme';
import { CreateAccount, mapDispatchToProps } from './CreateAccount';
import React, { Component } from 'react';

describe('CreateAccount', () => {
  let history;
  let wrapper;
  beforeEach(() => {
    history = {
      push: jest.fn()
    }
    wrapper = shallow(<CreateAccount history={history}/>);
  })

  it('should match snapshot upon initial render', () => { 
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handleDataEntry on change to the form', () => {
    wrapper = mount(
        <CreateAccount history={history}/>
      );
    const spy = jest.spyOn(wrapper.instance(), 'handleDataEntry');
    wrapper.find('form').simulate('change');
    wrapper.find('form').simulate('change');
    expect(spy).toHaveBeenCalled();
  })

  it('handleDataEntry should update the state with input', () => {
    const initialState = {
      name: '',
      email: '',
      password: '',
      dateOfBirth: ''
    }
    const mockEvent = {
      target: {
        name: 'email',
        value: 'pat@pat.com'
      }
    }
    const expectedState = {
      name: '',
      email: 'pat@pat.com',
      password: '',
      dateOfBirth: ''
    }
    expect(wrapper.state()).toEqual(initialState);
    wrapper.instance().handleDataEntry(mockEvent);
    expect(wrapper.state()).toEqual(expectedState);
  })

  // it('should call handleCreateAccount on form submit', () => {
  //   const mockHandleCreateAccount = jest.fn()
  //   wrapper = mount(
  //       <CreateAccount 
  //        handleCreateAccount={mockHandleCreateAccount}
  //        history={history}
  //       />);
  //   wrapper.find('form').simulate('submit');
  //   expect(mockHandleCreateAccount).toHaveBeenCalled();
  // })
})

describe('mapDispatchToProps', () => {

  it('handleCreateAccount should call dispatch', () => {
    const mockDispatch = jest.fn()
    const mappedProps = mapDispatchToProps(mockDispatch);
    const accountInfo = {
      name: 'bob'
    }
    mappedProps.handleCreateAccount(accountInfo);
    expect(mockDispatch).toHaveBeenCalled
  })
})