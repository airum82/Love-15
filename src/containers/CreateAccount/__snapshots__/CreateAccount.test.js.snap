// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount should match snapshot upon initial render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateAccount />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onChange={[Function]}
        onSubmit={[Function]}
>
        <input
                name="email"
                type="email"
        />
        <input
                name="name"
                type="text"
        />
        <input
                name="password"
                type="password"
        />
        <input
                name="dateOfBirth"
                type="text"
        />
        <button>
                Create Account
        </button>
</form>,
      "className": "create-account",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="email"
            type="email"
/>,
          <input
            name="name"
            type="text"
/>,
          <input
            name="password"
            type="password"
/>,
          <input
            name="dateOfBirth"
            type="text"
/>,
          <button>
            Create Account
</button>,
        ],
        "onChange": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "dateOfBirth",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onChange={[Function]}
          onSubmit={[Function]}
>
          <input
                    name="email"
                    type="email"
          />
          <input
                    name="name"
                    type="text"
          />
          <input
                    name="password"
                    type="password"
          />
          <input
                    name="dateOfBirth"
                    type="text"
          />
          <button>
                    Create Account
          </button>
</form>,
        "className": "create-account",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              type="email"
/>,
            <input
              name="name"
              type="text"
/>,
            <input
              name="password"
              type="password"
/>,
            <input
              name="dateOfBirth"
              type="text"
/>,
            <button>
              Create Account
</button>,
          ],
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "dateOfBirth",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Account",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
